### PR TITLE
WIP of webstats

### PR DESCRIPTION
**Pull request by XCMer**
Not functional yet, but has gone through some refactoring. Trying out some concepts (and learning in the process too!)

### CODE REVIEW

**Comment by patmaddox** [comment]
Go you!

File: 2_adventures/001/xcmer/04_challenge_stats_web/webstats.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    Dir.entries(@path).select {|entry|
+      next if (entry == '.' || entry == '..' || entry == 'README.md')
+
+      calculate_user(entry, @path + entry)
+    }
+  end
+
+  def calculate_user(user, user_path)
+    Dir.entries(user_path).select {|entry|
+      next if (entry == '.' || entry == '..' || entry == 'README.md')
```
Comment (kotp) [inline 2_adventures/001/xcmer/04_challenge_stats_web/webstats.rb:101]: next if %w[. .. README.md].any? { |item| entry == item }`

Untested code... but may work.  Better if this is reused in multiple places to name those items, and reuse it.

Or even make a method that checks for them and then you can say `next if unwanted?(entry)`

**Comment by kotp** [comment]
LGTM

**Comment by patmaddox** [comment]
very cool! I love the nice clean simple Ruby :)

**Pull request merged**